src/main/java/de/doubleslash/keeptime/common/ScreenPosHelper.java

```diff
@@ -1,94 +1,109 @@
 
-   public ScreenPosHelper(final double absolutX, final double absolutY) {
-      this.absolutX = absolutX;
-      this.absolutY = absolutY;
-      calcRelativScreenpos();
-   }
+   private int screenHash;
 
-   public ScreenPosHelper(final int screenhash, final double xProportion, final double yProportion) {
-      this.screenhash = screenhash;
-      this.xProportion = xProportion;
-      this.yProportion = yProportion;
-      calcAbsolutScreenpos();
-   }
+   private double absoluteX;
+   private double absoluteY;
 
-   public double getAbsolutX() {
-      return absolutX;
-   }
+   private double proportionalX;
+   private double proportionalY;
 
-   public void setAbsolutX(final double absolutX) {
-      this.absolutX = absolutX;
-      calcRelativScreenpos();
+   public ScreenPosHelper(final int screenhash, final double proportionalX, final double proportionalY) {
+      this.screenHash = screenhash;
+      this.proportionalX = proportionalX;
+      this.proportionalY = proportionalY;
+      calcAbsolutePosition();
    }
 
-   public double getAbsolutY() {
-      return absolutY;
+   public double getAbsoluteX() {
+      return absoluteX;
    }
 
-   public void setAbsolutY(final double absolutY) {
-      this.absolutY = absolutY;
-      calcRelativScreenpos();
+   public void setAbsoluteX(final double absolutX) {
+      this.absoluteX = absolutX;
+      calcProportionalPosition();
    }
 
-   public int getScreenhash() {
-      return screenhash;
+   public double getAbsoluteY() {
+      return absoluteY;
    }
 
-   public void setScreenhash(final int screenhash) {
-      this.screenhash = screenhash;
-      calcRelativScreenpos();
+   public void setAbsoluteY(final double absolutY) {
+      this.absoluteY = absolutY;
+      calcProportionalPosition();
    }
 
-   public double getxProportion() {
-      return xProportion;
+   public int getScreenHash() {
+      return screenHash;
    }
 
-   public void setxProportion(final double xProportion) {
-      this.xProportion = xProportion;
-      calcAbsolutScreenpos();
+   public double getProportionalX() {
+      return proportionalX;
    }
 
-   public double getyProportion() {
-      return yProportion;
+   public double getProportionalY() {
+      return proportionalY;
    }
 
-   public void setyProportion(final double yProportion) {
-      this.yProportion = yProportion;
-      calcAbsolutScreenpos();
+   public void resetPositionIfInvalid() {
+      if (!isPositionValid()) {
+         final Screen screen = getScreenWithHashOrPrimary(this.screenHash);
+         final Rectangle2D screenBounds = screen.getBounds();
+         LOG.warn(
+               "Position is not in the range of the screen. Screen (hash '{}') range '{}'. Calculated positions '{}'/'{}'. Setting to '0'/'0'.",
+               screen.hashCode(), screenBounds, absoluteX, absoluteY);
+         proportionalX = 0;
+         proportionalY = 0;
+         absoluteX = 0;
+         absoluteY = 0;
```

Comment:
would set proportional to 0.5 and then calculate the absolute to place in the middle  of the screen